pytorch_lightning/callbacks/quantization.py

```diff
@@ -217,3 +266,28 @@ def on_fit_end(self, trainer, pl_module):
             pl_module.forward = wrap_quantize_forward_context(model=pl_module, func=self.__module_forward)
         else:
             pl_module.forward = self.__module_forward
+
+    def on_validation_start(self, trainer, pl_module):
+        self._maybe_disable_observer(pl_module, "validate" in self._disable_observers and not trainer.sanity_checking)
+
+    def on_validation_end(self, trainer, pl_module):
+        if "validate" in self._disable_observers:
+            if trainer.sanity_checking:
+                for fake_quant, state_dict in self._fake_quant_to_initial_state_dict.items():
+                    fake_quant.load_state_dict(state_dict)
+            else:
+                self._restore_last_observer_enabled()
+
+    def on_test_start(self, trainer, pl_module):
+        self._maybe_disable_observer(pl_module, "test" in self._disable_observers)
+
+    def on_test_end(self, trainer, pl_module):
+        if "test" in self._disable_observers:
```

Comment:
why not move it to the `_restore` the same as you have `_maybe`

Reply:
Passing `should_disable` to the method seems strange for me today :(. Maybe we don't need to extract `if` to the method in this case. I remove the `_maybe` from `QuantizationAwareTraining` callbacks.